include/openthread/platform/settings.h

```diff
@@ -67,6 +90,18 @@ void otPlatSettingsInit(otInstance *aInstance);
  */
 void otPlatSettingsDeinit(otInstance *aInstance);
 
+/**
+ * This function sets the critical keys that should be stored in the secure area.
+ *
+ * Note that the memory pointed by @p aKeys MUST not be released before @p aInstance is destroyed.
+ *
+ * @param[in]  aInstance    The OpenThread instance structure.
+ * @param[in]  aKeys        A pointer to an array containing the list of critical keys.
+ * @param[in]  aKeysLength  The number of entries in the @p aKeys array.
+ *
+ */
+void otPlatSettingsSetCriticalKeys(otInstance *aInstance, const uint16_t *aKeys, uint16_t aKeysLength);
```

Comment:
nit: 
Why do we still need this method if we have exposed all keys? @bukepo @zhanglongxia 
The platform can freely decide which keys are critical?

Reply:
You can refer to the @abtink's comments above https://github.com/openthread/openthread/pull/6125#issuecomment-771775435.